### PR TITLE
Delete a file from Drive

### PR DESCRIPTION
**Pull request by Ninja243**
Lets the user delete a single file from Drive.

### CODE REVIEW

File: deta/base.py [hunk 1]
```diff
@@ -100,10 +100,22 @@ def __init__(self, drive_name:str=None, project_key:str=None, project_id:str=Non
         self.project_key = project_key
         self.project_id = project_id
         host = host or os.getenv("DETA_DRIVE_HOST") or "drive.deta.sh"
-        self.client = http.client.HTTPSConnection(host, timeout=3)
+        self.client = http.client.HTTPConnection(host, timeout=300)
```
Comment (aavshr) [inline deta/base.py]: should be `https`.

File: deta/base.py [hunk 2]
```diff
@@ -100,10 +100,22 @@ def __init__(self, drive_name:str=None, project_key:str=None, project_id:str=Non
...
+        name: the name of item to be deleted
+        """
+        if name == "":
+            raise ValueError("Name is empty")
+        # encode key
+        key = quote(name, safe="")
+        status, payload = self._request("/files", "DELETE", {"names":[name]})
+        if (status == 404):
+            return None
+        return payload["deleted"]
```
Comment (aavshr) [inline deta/base.py:140]: delete returns a `2xx` status code even if some deletion fails and the failed files are returned in `payload['failed']`

we need to raise an error if deletion fails here

File: deta/base.py (hunk 1 above)
Comment (Ninja243) [reply deta/base.py]: Thanks!

File: deta/base.py (hunk 2 above)
Comment (Ninja243) [reply deta/base.py:140]: Gotcha, thanks!

**Pull request merged**